### PR TITLE
Enthought Sphinx Theme for Docs

### PR DESCRIPTION
**Pull request by jjenthought**
This mirrors the documentation setup for TraitsUI, where we try to use the [enthought-sphinx-theme](https://github.com/enthought/enthought-sphinx-theme) and fall back to the older defaults if this is not available. Though this presumably should only happen for someone building the docs locally, since anyone building them to upload to the website should probably just download the enthought theme!

This also fixes a bug where the docs would not build without a version of traitsui. This was due to the `DocMock` class calling `getattr(self, _mock_methods)` recursively on initialisation. 

There are some new warnings due to Traits having classes exclusive to python 2 - however sphinx effectively ends up just skipping these rather than it causing any problems (as far as I can tell anyway!). These warnings are also present if the docs are built with traitsui present on python 3.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -191,33 +191,49 @@ def __name__(self):
...
+    # Use old defaults if enthought-sphinx-theme not available
+
+    # The name of an image file (within the static path) to place at the top
+    # of the sidebar.
+    html_logo = "e-logo-rev.png"
+
+    # The name of an image file (within the static path) to use as favicon of
+    # the docs.  This file should be a Windows icon file (.ico) being 16x16
+    # or 32x32 pixels large.
+    html_favicon = "et.ico"
```
Comment (mdickinson) [inline docs/source/conf.py]: I'm getting a warning: `WARNING: favicon file 'et.ico' does not exist` here. Are we setting the path to the static files correctly?

Comment (mdickinson) [reply docs/source/conf.py]: Okay, this isn't new in this PR; it's an issue on the master branch, too.

**Comment by mdickinson** [comment]
Tested both with and without the enthought sphinx theme. LGTM.

File: docs/source/conf.py (hunk 1 above)
Comment (jjenthought) [reply docs/source/conf.py]: Just out of interest, when does the warning come up? I couldn't see it when I was building the html docs just now, which is a bit strange.

**Pull request merged**